app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/MapViewModel.java

```diff
@@ -0,0 +1,274 @@
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Handler;
+import android.widget.Toast;
+
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.common.logging.LoggerFactory;
+import com.teamagam.gimelgimel.app.control.sensors.LocationFetcher;
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageBroadcastReceiver;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageMapEntitiesViewModel;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageUserLocation;
+import com.teamagam.gimelgimel.app.model.ViewsModels.UsersLocationViewModel;
+import com.teamagam.gimelgimel.app.model.entities.LocationSample;
+import com.teamagam.gimelgimel.app.network.services.GGImageSender;
+import com.teamagam.gimelgimel.app.network.services.IImageSender;
+import com.teamagam.gimelgimel.app.utils.Constants;
+import com.teamagam.gimelgimel.app.view.MainActivity;
+import com.teamagam.gimelgimel.app.map.view.ViewerFragment;
+import com.teamagam.gimelgimel.app.view.fragments.dialogs.SendMessageDialogFragment;
+import com.teamagam.gimelgimel.app.map.view.GGMapView;
+import com.teamagam.gimelgimel.app.map.model.VectorLayer;
+import com.teamagam.gimelgimel.app.map.model.entities.Entity;
+import com.teamagam.gimelgimel.app.map.model.entities.Point;
+import com.teamagam.gimelgimel.app.map.model.geometries.PointGeometry;
+import com.teamagam.gimelgimel.app.map.model.symbols.PointImageSymbol;
+import com.teamagam.gimelgimel.app.map.model.symbols.PointSymbol;
+import com.teamagam.gimelgimel.domain.base.logging.Logger;
+import com.teamagam.gimelgimel.presentation.scopes.PerFragment;
+
+import javax.inject.Inject;
+
+/**
+ * View Model that handles map related callbacks from user in {@link ViewerFragment}.
+ * Also handles it's view {@link GGMapView}.
+ * <p>
+ * Controls communication between views and models of the presentation
+ * layer.
+ */
+@PerFragment
+public class MapViewModel {
+//implements SendGeographicMessageDialog.SendGeographicMessageDialogInterface
+
+    IMapView mMapView;
+
+    private VectorLayer mSentLocationsLayer;
+    private VectorLayer mUsersLocationsLayer;
+    private VectorLayer mReceivedLocationsLayer;
+
+    private MessageBroadcastReceiver mUserLocationReceiver;
+    private BroadcastReceiver mLocationReceiver;
+    private IImageSender mImageSender;
+
+    private Uri mImageUri;
+    private boolean mIsRestored;
+    private Handler mHandler;
+    private Runnable mPeriodicalUserLocationsRefreshRunnable;
+
+
+    //injects
+    @Inject
+    MessageMapEntitiesViewModel mMessageLocationVM;
+
+    private UsersLocationViewModel mUserLocationsVM;
+
+    private final MainActivity mainActivity;
+
+    Context mContext;
+
+    //logger
+    private Logger sLogger = LoggerFactory.create(getClass());
+
+    @Inject
+    public MapViewModel(Context context, MainActivity activity, UsersLocationViewModel userLocationVM) {
```

Comment:
Extract the constructor code to methods.

You may not extract the members assignment code lines.